src/main/java/com/synopsys/integration/detect/lifecycle/run/operation/OperationRunner.java

```diff
@@ -346,20 +353,80 @@ public UUID initiateStatelessBdbaScan(BlackDuckRunData blackDuckRunData) throws
     }
 
 
-    private String getScanServicePostEndpoint() {
+    public String getScanServicePostEndpoint() {
         if (detectConfigurationFactory.createScanMode() == BlackduckScanMode.INTELLIGENT) {
             return INTELLIGENT_SCAN_ENDPOINT;
         }
         return RAPID_SCAN_ENDPOINT;
     }
 
-    private String getScanServicePostContentType() {
+    public String getScanServicePostContentType() {
         if (detectConfigurationFactory.createScanMode() == BlackduckScanMode.INTELLIGENT) {
             return INTELLIGENT_SCAN_CONTENT_TYPE;
         }
         return RAPID_SCAN_CONTENT_TYPE;
     }
 
+    public File getContainerScanImage() {
+        Optional<Path> containerImageFilePath = detectConfigurationFactory.getContainerScanFilePath();
+        File containerImageFile = null;
+        if (containerImageFilePath.isPresent()) {
+            containerImageFile = containerImageFilePath.get().toFile();
```

Comment:
NPE check may be needed.

Reply:
Yes, NPE check is done by `containerImageFilePath.isPresent()`